pkg/ui/bucket.go

```diff
@@ -48,28 +48,33 @@ func NewBucketUI(logger log.Logger, label, externalPrefix, prefixHeader, uiPrefi
 }
 
 // Register registers http routes for bucket UI.
-func (b *Bucket) Register(r *route.Router, registerNewUI bool, ins extpromhttp.InstrumentationMiddleware) {
-	instrf := func(name string, next func(w http.ResponseWriter, r *http.Request)) http.HandlerFunc {
-		return ins.NewHandler(b.externalPrefix+name, http.HandlerFunc(next))
-	}
-	r.WithPrefix(b.uiPrefix).Get("/", instrf("root", b.root))
-	r.WithPrefix(b.uiPrefix).Get("/static/*filepath", instrf("static", b.serveStaticAsset))
-	if registerNewUI {
-		// Make sure that "<path-prefix>/new" is redirected to "<path-prefix>/new/" and
-		// not just the naked "/new/", which would be the default behavior of the router
-		// with the "RedirectTrailingSlash" option (https://godoc.org/github.com/julienschmidt/httprouter#Router.RedirectTrailingSlash),
-		// and which breaks users with a --web.route-prefix that deviates from the path derived
-		// from the external URL.
-		r.Get("/new", func(w http.ResponseWriter, r *http.Request) {
-			http.Redirect(w, r, path.Join(GetWebPrefix(b.logger, b.externalPrefix, b.prefixHeader, r), "new")+"/", http.StatusFound)
-		})
-		r.Get("/new/*filepath", instrf("react-static", b.serveReactUI))
+func (b *Bucket) Register(r *route.Router, ins extpromhttp.InstrumentationMiddleware) {
+	redirectPath := "/blocks"
+	if len(b.uiPrefix) > 0 {
+		redirectPath = b.uiPrefix
 	}
+	r.Get("/", func(w http.ResponseWriter, r *http.Request) {
+		http.Redirect(w, r, path.Join(GetWebPrefix(b.logger, b.externalPrefix, b.prefixHeader, r), redirectPath), http.StatusFound)
+	})
+	// Redirect the original React UI's path (under "/new") to its new path at the root.
+	r.Get("/new/*path", func(w http.ResponseWriter, r *http.Request) {
+		p := route.Param(r.Context(), "path")
+		http.Redirect(w, r, path.Join(GetWebPrefix(b.logger, b.externalPrefix, b.prefixHeader, r), strings.TrimPrefix(p, "/new"))+"?"+r.URL.RawQuery, http.StatusFound)
+	})
+
+	// here we have two routes that serve the same document. It's because it depends where do we come from.
+	// If we are coming from the new UI, it will use the first route.
+	// If we are coming from the old UI, it will use the second route.
+	r.Get(path.Join("/classic", b.uiPrefix), instrf("bucket", ins, b.bucket))
+	r.WithPrefix(b.uiPrefix).Get("/classic", instrf("bucket", ins, b.bucket))
+	r.WithPrefix(b.uiPrefix).Get("/classic/static/*filepath", instrf("static", ins, b.serveStaticAsset))
```

Comment:
this part is quite ugly, but, I fixed every possible path that you could follow. So we can now move from the old UI to the new one without any issue. 

But definitely dropping the old UI will simplify a lot the way to serve the UI.

I didn't really find the good way to have the same route everywhere. It's mainly becaused the prefix of the UI is not managed in the same way on the old UI and in the UI. (At least that's what I am seing and what I understood)